packages/flutter/lib/src/widgets/layout_builder.dart

```diff
@@ -256,42 +276,47 @@ class _LayoutBuilderElement<ConstraintType extends Constraints> extends RenderOb
 
   @override
   void removeRenderObjectChild(RenderObject child, Object? slot) {
-    final RenderConstrainedLayoutBuilder<ConstraintType, RenderObject> renderObject =
+    final RenderConstrainedLayoutBuilder<LayoutInfoType, RenderObject> renderObject =
         this.renderObject;
     assert(renderObject.child == child);
     renderObject.child = null;
     assert(renderObject == this.renderObject);
   }
 }
 
-/// Generic mixin for [RenderObject]s created by [ConstrainedLayoutBuilder].
+/// Generic mixin for [RenderObject]s created by [AbstractLayoutBuilder].
```

Comment:
I am a little confused by this class. Is this supposed to work with generic `LayoutInfoType`s (like `AbstractLayoutBuilder` does) or with `Constraints` specifically? The doc here and the `LayoutInfoType` generic argument seem to imply its supposed to be generic. But then the name (RenderConstrainedLayoutBuilder) and the fact that it hardcodes `Constraints` in multiple places below seem to indicate the opposite. Could you clarify this and make it clear whether it is one or the other? Or make it clearer why this is both?

Reply:
I think it's the former.  Added a lot more text to the class / method documentation.

Reply:
I'll try to rename this to `AbstractLayoutBuilderMixin`. Hopefully I can get away with a rename.

Reply:
Looks like all registered tests passed.